### PR TITLE
add: partial import with progress bar

### PR DESCRIPTION
**Pull request by omidrezakeshtkar**

### CODE REVIEW

File: src/utils/shared/headers.ts [hunk 1]
```diff
@@ -57,3 +57,33 @@ export type englishHeader = {
...
+        degree: string,
+        grade: string,
+        schoolName: string,
+        schoolType: string,
+        schoolCode: string,
+        schoolGender: string,
+        mobileNumber: string,
+        parentMobileNumber: string
+    },
+    enabled: string | undefined
```
[the remark this answers is not in the document]
Comment (omidrezakeshtkar) [reply src/utils/shared/headers.ts:88]: Users can be set to enabled or disabled by default, which is indicated by enabled.

File: src/utils/requestModule/getUsersList.ts [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+export const getUsersListFromKeycloak= async(accessToken: string)=>{
     const response = await axios({
         method: 'get',
         url: `${process.env.keycloak_address}/auth/admin/realms/${process.env.keycloak_realm}/users`,
         headers: { Authorization: accessToken }
     })
     return response.data
+}
+
+const exportUsernameList=(data: string[])=>{
```
Comment (PedramRZM) [inline src/utils/requestModule/getUsersList.ts:14]: please use prettier to reformat your code.

File: src/utils/requestModule/shared/duplicate.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import * as lodash from 'lodash'
+import { usernameListType } from "utils/shared"
+
+export const duplicateCheck = async(usernamelists: Promise<usernameListType>): Promise<string[]> => {
```
Comment (PedramRZM) [inline src/utils/requestModule/shared/duplicate.ts:4]: I think there is no need to wrap this logic around a function. especially `removeDuplicateData`

File: src/utils/requestModule/getUsersList.ts (hunk 2 above)
Comment (PedramRZM) [inline src/utils/requestModule/getUsersList.ts:14]: No need to put this into a function

File: src/utils/requestModule/postUsersImport.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+
+export const postSingleUsersImport = async (accessToken:string, jsondata: object) => {
```
Comment (PedramRZM) [inline src/utils/requestModule/postUsersImport.ts:3]: please rename this file to function name. I think this file is redundant and can be deleted.

File: src/utils/requestModule/requestModule.ts [hunk 5]
```diff
@@ -1,30 +1,40 @@
...
-                    data: element,
-                })
-            } catch (error) {
-                console.log(error);
-            }
-        }else{
-            warningLog('duplicate user: ', element.username)
-        }
-    })   
+    const requestdataarray: string[] = convertJsonDataToKeycloakFormat(jsonarray)
```
Comment (PedramRZM) [inline src/utils/requestModule/requestModule.ts:18]: please use camelCase for your variables in the whole codebase

File: src/utils/requestModule/requestModule.ts [hunk 6]
```diff
@@ -1,30 +1,40 @@
...
+    const usernamelists: Promise<usernameListType> = getUserNameList(requestdataarray)
+    const duplicateusernamelist: string[] = await duplicateCheck(usernamelists)
+    const requestarray:string[] = removeDuplicateData(requestdataarray, duplicateusernamelist)
+    const batchUsers: string[][] = lodash.chunk(requestarray, 1000)
```
Comment (PedramRZM) [inline src/utils/requestModule/requestModule.ts:22]: you can put 1000 into env or config files

File: src/utils/csvModule/convertJsonData.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
-import { englishHeader } from "utils/shared";
+import { englishHeader, keycloakUserFormat } from "utils/shared";
 
-export const convertJsonDataToKeycloakFormat=(jsonarray: any[])=>{
-    return jsonarray.map((element: englishHeader)=>{
+export const convertJsonDataToKeycloakFormat=(jsonarray: any[]): any[]=>{
```
Comment (PedramRZM) [inline src/utils/csvModule/convertJsonData.ts:3]: no wrapper function is needed

**Review by PedramRZM — COMMENTED** [review]
reviewed

**Pull request merged**